x-pack/plugins/fleet/public/applications/fleet/sections/settings/components/edit_output_flyout/advanced_options_section.tsx

```diff
@@ -0,0 +1,303 @@
+          <EuiFlexGroup alignItems="flexStart">
+            <EuiFlexItem>
+              <EuiSwitch
+                data-test-subj="editOutputFlyout.diskQueueSwitch"
+                {...diskQueueEnabledInput.props}
+                label={
+                  <FormattedMessage
+                    id="xpack.fleet.settings.editOutputFlyout.diskQueueSwitchLabel"
+                    defaultMessage="Disk Queue"
+                  />
+                }
+              />
+            </EuiFlexItem>
+            <EuiFlexItem>
+              <EuiText>
+                <FormattedMessage
+                  id="xpack.fleet.settings.editOutputFlyout.diskQueueSwitchDescription"
+                  defaultMessage="Once enabled, events will be queued on disk when there's a requirement to cache them on the agents in the event of connection loss."
+                />
+              </EuiText>
+            </EuiFlexItem>
+          </EuiFlexGroup>
+        </EuiFormRow>
+        <EuiSpacer size="m" />
+
+        <EuiFormRow fullWidth {...diskQueueEncryptionEnabled.formRowProps}>
+          <EuiFlexGroup alignItems="flexStart">
+            <EuiFlexItem>
+              <EuiSwitch
+                data-test-subj="editOutputFlyout.diskQueueEncryption"
+                {...diskQueueEncryptionEnabled.props}
+                label={
+                  <FormattedMessage
+                    id="xpack.fleet.settings.editOutputFlyout.diskQueueEncryptionLabel"
+                    defaultMessage="Encryption"
+                  />
+                }
+              />
+            </EuiFlexItem>
+            <EuiFlexItem>
+              <EuiText>
+                <FormattedMessage
+                  id="xpack.fleet.settings.editOutputFlyout.diskQueueEncryptionDescription"
+                  defaultMessage="Enable encryption of data at rest."
+                />
+              </EuiText>
+            </EuiFlexItem>
+          </EuiFlexGroup>
+        </EuiFormRow>
+        <EuiSpacer size="m" />
+        <EuiFormRow
+          fullWidth
+          label={
+            <FormattedMessage
+              id="xpack.fleet.settings.editOutputFlyout.diskQueuePathLabel"
+              defaultMessage="Path"
+            />
+          }
+          {...diskQueuePathInput.formRowProps}
+        >
+          <EuiFieldText
+            fullWidth
+            data-test-subj="settingsOutputsFlyout.diskQueuePath"
+            {...diskQueuePathInput.props}
+            placeholder={i18n.translate(
+              'xpack.fleet.settings.editOutputFlyout.diskQueuePathPlaceholder',
+              {
+                defaultMessage: 'path_data/diskqueue',
+              }
+            )}
+          />
+        </EuiFormRow>
+
+        <EuiFormRow
+          fullWidth
+          {...diskQueueMaxSizeInput.formRowProps}
+          label={
+            <FormattedMessage
+              id="xpack.fleet.settings.editOutputFlyout.diskQueueMaxSize"
+              defaultMessage="Maximum Disk Size"
```

Comment:
We're describing the disk queue, not the disk here.

```suggestion
              defaultMessage="Maximum Disk Queue Size"
```